files/etc/ublue-update.d/system/00-system-update.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+
+from subprocess import run
+from json import loads, load
+from json.decoder import JSONDecodeError
+import os
+
+
+def check_for_rebase():
+    # initialize variables
+    default_image_ref = []
+    current_image_ref = []
+    default_image_tag = ""
+    image_tag = ""
+
+    try:
+        with open("/usr/share/ublue-os/image-info.json") as f:
+            image_info = load(f)
+            default_image_ref = image_info["image-ref"].split(":")
+            default_image_tag = image_info["image-tag"]
+    except (FileNotFoundError, KeyError):
+        print("uBlue image info file does not exist")
+        return False, ""
+
+    status_cmd = [
+        "rpm-ostree",
+        "status",
+        "--pending-exit-77",
+        "-b",
+        "--json",
+    ]
+    status_out = run(status_cmd, capture_output=True)
+    if status_out.returncode != 0:
+        print(status_out.stdout.decode("utf-8"))
+        return False, ""
+
+    try:
+        current_image_ref = (
+            loads(status_out.stdout)["deployments"][0]["container-image-reference"]
+            .replace(
+                "ostree-unverified-registry:", "ostree-unverified-image:docker://"
+            )  # replace shorthand
+            .split(":")
+        )
+        if current_image_ref == default_image_ref:
+            return False, ""
+    except (JSONDecodeError, KeyError):
+        print("unable to parse JSON output")
+        print(status_out.stdout.decode("utf-8"))
+        return False, ""
+
+    try:  # preserve image tag when rebasing unsigned
+        if current_image_ref[2] == default_image_ref[2]:
+            image_tag = current_image_ref[3]
+    except KeyError:
+        print("unable to get image tag from current deployment!")
+        image_tag = default_image_tag
+
+    return True, f"{image_ref[0]}:{image_ref[1]}:{image_ref[2]}:{image_tag}"
```

Comment:
the issue with the `check_for_rebase` function was resolved with: https://github.com/ublue-os/ublue-update/pull/81/commits/5e5b931d70f7ff29d1cb3369927c2d0edbe74e3b
however i'm curious about the rebase failing on the `ostree-unverified-image:docker://`

Reply:
I will do the same tests again when I have time. With the new version you committed and report back how it went.

Reply:
Ran the same tests as above with https://github.com/ublue-os/ublue-update/pull/81/commits/5e5b931d70f7ff29d1cb3369927c2d0edbe74e3b and the results are the same. The rebasing works without any problems in the situations above. 

The only thing is when you are on a `ostree-image-signed` image and you run the script it will output:
`rebase failed!, command output:`